packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,509 @@
+const String _kBodyID = 'body';
+const String _kSecondaryBodyID = 'secondaryBody';
+
+/// A parent Widget takes in multiple [SlotLayout] components and places them
+/// into their appropriate positions on the screen.
+class AdaptiveLayout extends StatefulWidget {
+  /// Creates an [AdaptiveLayout] widget.
+  const AdaptiveLayout({
+    this.primaryNavigation,
+    this.secondaryNavigation,
+    this.topNavigation,
+    this.bottomNavigation,
+    this.body,
+    this.secondaryBody,
+    this.bodyRatio,
+    this.internalAnimations = true,
+    this.horizontalBody = true,
+    super.key,
+  });
+
+  /// The slot placed on the beginning side of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? primaryNavigation;
+
+  /// The slot placed on the end side of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? secondaryNavigation;
+
+  /// The slot placed on the top part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? topNavigation;
+
+  /// The slot placed on the bottom part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? bottomNavigation;
+
+  /// The slot that fills the rest of the space in the center.
+  final SlotLayout? body;
+
+  /// A supporting slot for [body]. Has a sliding entrance animation by default.
+  /// The default ratio for the split between [body] and [secondaryBody] is so
+  /// that the split axis is in the center of the screen.
+  final SlotLayout? secondaryBody;
+
+  /// Defines the fractional ratio of [body] to the [secondaryBody].
+  ///
+  /// For example 0.3 would mean [body] takes up 30% of the available space
+  /// and[secondaryBody] takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
+  /// the center of the screen.
+  final double? bodyRatio;
+
+  /// Whether or not the developer wants the smooth entering slide transition on
+  /// [secondaryBody].
+  ///
+  /// Defaults to true.
+  final bool internalAnimations;
+
+  /// Whether to orient the body and secondaryBody in horizontal order (true) or
+  /// in vertical order (false).
+  ///
+  /// Defaults to true.
+  final bool horizontalBody;
+
+  @override
+  State<AdaptiveLayout> createState() => _AdaptiveLayoutState();
+}
+
+class _AdaptiveLayoutState extends State<AdaptiveLayout>
+    with TickerProviderStateMixin {
+  late AnimationController _controller;
+
+  late Map<String, SlotLayoutConfig?> chosenWidgets =
+      <String, SlotLayoutConfig?>{};
+  Map<String, Size?> slotSizes = <String, Size?>{
```

Comment:
I think all of these should be empty maps, and if a value isn't found in the map, then it is assumed to be zero. Then you wouldn't need all `updateAll` to set them back to zero to reset them, you could just call `.clear()` on the map.

Same for the equivalents in the delegate.